main.go

```diff
@@ -41,27 +46,31 @@ var (
 )
 
 func init() {
-	_ = clientgoscheme.AddToScheme(scheme)
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
-	_ = syncv1alpha1.AddToScheme(scheme)
-	_ = argov1alpha1.AddToScheme(scheme)
-	_ = applicationset.AddToScheme(scheme)
-	// +kubebuilder:scaffold:scheme
+	utilruntime.Must(syncv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(argov1alpha1.AddToScheme(scheme))
+	utilruntime.Must(applicationset.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
 }
 
 func main() {
 	var metricsAddr, ctrlNamespace, argoNamespace string
 	var enableLeaderElection bool
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.StringVar(&ctrlNamespace, "ctrl-namespace", "argocd", "The controller namespace")
 	flag.StringVar(&argoNamespace, "argo-namespace", "argocd", "The namespace where ArgoCD and the ApplicationSet controller are deployed to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-
-	opts := zap.Options{}
+	opts := zap.Options{
+		Development: true,
```

Comment:
that is going to generate tons of logs by default right? I'd rather we leave the line below add the dev flag on

Reply:
I _think_ that what changes is the log format. Looking at https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap@v0.6.4#Options it says 
```
	// Development configures the logger to use a Zap development config
	// (stacktraces on warnings, no sampling), otherwise a Zap production
	// config will be used (stacktraces on errors, sampling).
```
Doesn't sound too bad but I can change it back if we want

Reply:
I was under the impression it would activate all the `DEBUG` traces, i was prob wrong.
We can start like that and see what happens.